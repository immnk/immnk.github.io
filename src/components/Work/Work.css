.root {
}

.container {
  width: 720px;
  margin: 0 auto;
  padding: 8rem 2.4rem 16rem 2.4rem;

  @media screen and (max-width: 540px) {
    width: 100%;
  }
}

.header {
  font-family: 'Abel', sans-serif;
  font-weight: 800;
  font-size: 3.2rem;
  margin-bottom: 2.4rem;
  text-align: center;
}

.description {
  font-weight: 300;
  line-height: 1.5;
  font-size: 1.6rem;
  text-align: center;
}

.workListWrapper {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-auto-rows: 200px;
}

.title {
}

.work {
  position: relative;
  margin: 20px;

  .title {
    position: absolute;
    bottom: 0;
    left: 50%;
    transform: translateX(-50%);
    margin-bottom: 20px;
    font-size: 1.6rem;
    color: #fff;
  }

  &::before {
    content: '';
    position: absolute;
    left: 0;
    width: 100%;
    height: 100%;
    background: rgb(0, 60, 68);
    background: linear-gradient(
      0deg,
      rgba(0, 60, 68, 1) 0%,
      rgba(4, 81, 115, 0.83) 30%,
      rgba(33, 218, 255, 0) 100%
    );
  }
}

.image {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.companiesWorkedWrapper {
  .header {
    margin-top: 50px;
  }
}

.companies {
  display: grid;
  grid-gap: 10px;
  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
}

.company {
  position: relative;
  text-decoration: none;
  text-align: center;

  .image {
    height: 100px;
    object-fit: contain;
  }

  .title {
    color: #111;
    font-size: 1.4rem;
  }
}
