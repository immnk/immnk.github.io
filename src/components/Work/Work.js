import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import ScrollAnimation from 'react-animate-on-scroll';

import PranavamImage from './images/pranavam.png';
import CreatusLabImage from './images/creatuslabs.png';
import BotDesignersImage from './images/baashabot.jpeg';
import AryaWatsonImage from './images/aryawatson.jpeg';
import SiriusKartImage from './images/siriuskart.png';
import DefaultImage from './images/project.png';

import css from './Work.css';

const Work = ({ className }) => {
  const works = [
    {
      name: 'Pranavam Natural Life',
      image: PranavamImage,
      url: 'https://pranavamnaturallife.com/'
    },
    { name: 'Creatus labs', image: CreatusLabImage, url: 'https://www.behance.net/manikantamnk' },
    {
      name: 'The Bot Designers',
      image: BotDesignersImage,
      url: 'https://www.facebook.com/thebotdesigners'
    },
    { name: 'Judging App', url: 'https://judgingapp-admin.herokuapp.com/' },
    {
      name: 'Arya Watson',
      image: AryaWatsonImage,
      url: 'https://www.messenger.com/t/121496828551626'
    },
    {
      name: 'Sirius Kart',
      image: SiriusKartImage,
      url: 'https://www.messenger.com/t/296374397500984'
    }
  ];

  const companiesWorked = [
    {
      name: 'DBS Bank',
      image: 'https://www.dbs.com.sg/iwov-resources/flp/images/dbs_logo.svg',
      url: 'https://www.dbs.com.sg/'
    },
    {
      name: 'SGX',
      image: 'https://www.sgx.com/images/7b0d6cda8853ac989d6e8670592a46c6.svg',
      url: 'https://www.sgx.com/'
    },
    {
      name: 'Sirius',
      image: 'https://www.siriuscom.com/wp-content/uploads/2017/01/logoweb17.png',
      url: 'https://www.siriuscom.com/'
    }
  ];

  return (
    <section className={classNames(css.root, className)}>
      <ScrollAnimation animateIn="fadeIn" delay={100} initiallyVisible={false}>
        <div className={css.container}>
          <h3 className={css.header}>Previous Work</h3>
          <p className={css.description}>Here are list of my previous work done.</p>
          <div className={css.workListWrapper}>
            {works.map((work) => {
              return (
                <a
                  className={css.work}
                  href={work.url}
                  rel="noreferrer"
                  target="_blank"
                  key={work.name}
                >
                  <img alt={work.name} src={work.image || DefaultImage} className={css.image} />
                  <h4 className={css.title}>{work.name}</h4>
                </a>
              );
            })}
          </div>
          <div className={css.companiesWorkedWrapper}>
            <h4 className={css.header}>I&apos;m proud to work at these companies</h4>
            <div className={css.companies}>
              {companiesWorked.map((company) => {
                return (
                  <a
                    className={css.company}
                    href={company.url}
                    rel="noreferrer"
                    target="_blank"
                    key={company.name}
                  >
                    <img
                      alt={company.name}
                      src={company.image || DefaultImage}
                      className={css.image}
                    />
                    <h4 className={css.title}>{company.name}</h4>
                  </a>
                );
              })}
            </div>
          </div>
        </div>
      </ScrollAnimation>
    </section>
  );
};

Work.defaultProps = {
  className: undefined
};

Work.propTypes = {
  className: PropTypes.string
};

export default Work;
