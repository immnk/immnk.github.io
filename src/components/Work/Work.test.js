import React from 'react';
import { shallow } from 'enzyme';

import Work from './Work';

const renderWork = (props) => {
  const defaultProps = {
    className: 'my-classname'
  };

  return <Work {...defaultProps} {...props} />;
};

const shallowRenderWork = (props) => shallow(renderWork(props));

describe('Test Work', () => {
  it('render app', () => {
    const actual = shallowRenderWork();
    expect(actual).toMatchSnapshot();
  });
});
