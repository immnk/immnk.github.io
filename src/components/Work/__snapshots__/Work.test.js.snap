// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Work render app 1`] = `
<section
  className="root my-classname"
>
  <ScrollAnimation
    animateIn="fadeIn"
    animateOnce={false}
    animatePreScroll={true}
    delay={100}
    duration={1}
    initiallyVisible={false}
    offset={150}
  >
    <div
      className="container"
    >
      <h3
        className="header"
      >
        Previous Work
      </h3>
      <p
        className="description"
      >
        Here are list of my previous work done.
      </p>
      <div
        className="workListWrapper"
      >
        <a
          className="work"
          href="https://pranavamnaturallife.com/"
          key="Pranavam Natural Life"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Pranavam Natural Life"
            className="image"
            src="test-file-stub"
          />
          <h4
            className="title"
          >
            Pranavam Natural Life
          </h4>
        </a>
        <a
          className="work"
          href="https://www.behance.net/manikantamnk"
          key="Creatus labs"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Creatus labs"
            className="image"
            src="test-file-stub"
          />
          <h4
            className="title"
          >
            Creatus labs
          </h4>
        </a>
        <a
          className="work"
          href="https://www.facebook.com/thebotdesigners"
          key="The Bot Designers"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="The Bot Designers"
            className="image"
            src="test-file-stub"
          />
          <h4
            className="title"
          >
            The Bot Designers
          </h4>
        </a>
        <a
          className="work"
          href="https://judgingapp-admin.herokuapp.com/"
          key="Judging App"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Judging App"
            className="image"
            src="test-file-stub"
          />
          <h4
            className="title"
          >
            Judging App
          </h4>
        </a>
        <a
          className="work"
          href="https://www.messenger.com/t/121496828551626"
          key="Arya Watson"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Arya Watson"
            className="image"
            src="test-file-stub"
          />
          <h4
            className="title"
          >
            Arya Watson
          </h4>
        </a>
        <a
          className="work"
          href="https://www.messenger.com/t/296374397500984"
          key="Sirius Kart"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Sirius Kart"
            className="image"
            src="test-file-stub"
          />
          <h4
            className="title"
          >
            Sirius Kart
          </h4>
        </a>
      </div>
      <div
        className="companiesWorkedWrapper"
      >
        <h4
          className="header"
        >
          I'm proud to work at these companies
        </h4>
        <div
          className="companies"
        >
          <a
            className="company"
            href="https://www.dbs.com.sg/"
            key="DBS Bank"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="DBS Bank"
              className="image"
              src="https://www.dbs.com.sg/iwov-resources/flp/images/dbs_logo.svg"
            />
            <h4
              className="title"
            >
              DBS Bank
            </h4>
          </a>
          <a
            className="company"
            href="https://www.sgx.com/"
            key="SGX"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="SGX"
              className="image"
              src="https://www.sgx.com/images/7b0d6cda8853ac989d6e8670592a46c6.svg"
            />
            <h4
              className="title"
            >
              SGX
            </h4>
          </a>
          <a
            className="company"
            href="https://www.siriuscom.com/"
            key="Sirius"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="Sirius"
              className="image"
              src="https://www.siriuscom.com/wp-content/uploads/2017/01/logoweb17.png"
            />
            <h4
              className="title"
            >
              Sirius
            </h4>
          </a>
        </div>
      </div>
    </div>
  </ScrollAnimation>
</section>
`;
